code/modules/mob/mob_movement.dm

```diff
@@ -185,8 +185,9 @@ CLIENT_VERB(drop_item)
 		if(mob.next_move_slowdown)
 			move_delay += mob.next_move_slowdown
 			mob.next_move_slowdown = 0
-		if((mob.flags_atom & DIRLOCK) && mob.dir != direct)
-			move_delay += MOVE_REDUCTION_DIRECTION_LOCKED // by Geeves
+		if(!ishuman(mob)) //humans can dirlock with no slowdown
+			if((mob.flags_atom & DIRLOCK) && mob.dir != direct)
+				move_delay += MOVE_REDUCTION_DIRECTION_LOCKED // by Geeves
```

Comment:
Rather than typechecking can you make a var or mob flag for no slowdown

Reply:
done as a var